storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -2125,11 +2143,17 @@ bool copy_back(int argc, char **argv) {
     int ret = fscanf(f, "%u", &key);
     ut_a(ret == 1);
     fclose(f);
-
-    if (!xb_keyring_init_for_copy_back(argc, argv)) {
+    if(!xtrabackup::components::keyring_init_offline())
+    {
+      msg("xtrabackup: Error: failed to init keyring component\n");
+      return (false);
+    }
+    if (!xtrabackup::components::keyring_component_initialized &&
+        !xb_keyring_init_for_copy_back(argc, argv)) {
```

Comment:
can we refactor this a bit? move everything inside the init_for_copy_back() ?

Reply:
move this xtrabackup::components::intitialize_service_handles() inside copy_back() too